test/spec/proxyBenchmark.js

```diff
@@ -58,69 +43,153 @@ function generateBigObjectFixture(carsSize) {
   return obj;
 }
 
-suite.add('jsonpatcherproxy generate operation', function() {
-  const obj = generateDeepObjectFixture();
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  const patches = jsonPatcherProxy.generate();
-  observedObj.firstName = 'Joachim';
-  observedObj.lastName = 'Wester';
-  observedObj.phoneNumbers[0].number = '123';
-  observedObj.phoneNumbers[1].number = '456';
-});
-
-suite.add('jsonpatch generate operation', function() {
-  const obj = generateDeepObjectFixture();
-  const observer = jsonpatch.observe(obj);
+function modifyObj(obj) {
   obj.firstName = 'Joachim';
   obj.lastName = 'Wester';
-  obj.phoneNumbers[0].number = '123';
-  obj.phoneNumbers[1].number = '456';
-  jsonpatch.generate(observer);
-});
-
-suite.add('jsonpatcherproxy mutation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  observedObj.cars[50].name = 'Toyota'
-});
-
-suite.add('jsonpatch mutation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const observer = jsonpatch.observe(obj);
-  obj.cars[50].name = 'Toyota';
-  jsonpatch.generate(observer);
-});
-
-suite.add('jsonpatcherproxy generate operation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  observedObj.cars.shift();
-});
-
-suite.add('jsonpatch generate operation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const observer = jsonpatch.observe(obj);
-  obj.cars.shift();
-  jsonpatch.generate(observer);
-});
-
-suite.add('PROXIFY big object', function() {
-  const obj = generateBigObjectFixture(50);
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  observedObj.a = 1;
-});
-
-suite.add('DIRTY-OBSERVE big object', function() {
-  const obj = generateBigObjectFixture(50);
-  const observer = jsonpatch.observe(obj);
-  obj.a = 1;
-  jsonpatch.generate(observer);
-
-});
+  obj.cars[0].speed = 123;
+  obj.cars[0].temp.speed = 456;
+}
+
+function reverseString(str) {
+  return str.split("").reverse().join("");
+}
+
+/* ============================= */
+
+{
+  const suiteName = 'Observe and generate, small object';
+  
+  {
+    suite.add(`${suiteName} (noop)`, function() {
+      const obj = generateBigObjectFixture(1);
```

Comment:
Shouldn't generation of obj be moved outside of suite? (here and bellow?)

Reply:
Ah, it shouldn't as we want to proxify new object every time..

Reply:
> Ah, it shouldn't as we want to proxify new object every time..

Yes